imager/tests.py

```diff
@@ -0,0 +1,194 @@
+        response = self.client.get('/accounts/login/')
+        self.assertIn('<a href="/accounts/register/">Register</a>!',
+                      response.content)
+        self.assertIn('<input type="submit" value="Log in" />',
+                      response.content)
+
+    def test_logged_out_registration_page(self):
+        response = self.client.get('/accounts/register/')
+        self.assertIn('<h1>Register Here</h1>',
+                      response.content)
+
+    def test_logged_out_profile(self):
+        response = self.client.get('/profile/')
+        self.assertEqual(response.items()[3][1],
+                         'http://testserver/accounts/login/?next=/profile/')
+
+    def test_logged_out_stream(self):
+        response = self.client.get('/stream/')
+        self.assertEqual(response.items()[3][1],
+                         'http://testserver/accounts/login/?next=/stream/')
+
+    def test_logged_out_library(self):
+        response = self.client.get('/library/')
+        self.assertEqual(response.items()[3][1],
+                         'http://testserver/accounts/login/?next=/library/')
+
+
+class LoggedInTestCase(TestCase):
+    def setUp(self):
+        self.client = Client()
+        self.username = 'test_username'
+        UserFactory(username=self.username)
+        self.client.login(username=self.username, password=PASSWORD)
+        self.new_user = 'username'
+        self.new_password = 'password'
+        self.new_email = 'user@test.com'
+
+    def tearDown(self):
+        for file in glob.glob("media/imager_user/test*"):
+            os.remove(file)
+
+    def test_login_redirect_success(self):
+        UserFactory()
+        response = self.client.post('/accounts/login/',
+                                    {'username': 'test_username',
+                                     'password': PASSWORD})
+        self.assertRedirects(response, 'profile/')
+
+    def test_login_redirect_failuse(self):
+        UserFactory()
+        response = self.client.post('/accounts/login/',
+                                    {'username': 'wrong',
+                                     'password': 'wrong'})
+        self.assertIn('Please enter a correct username and password',
+                      response.content)
+
+    def test_logged_in_home(self):
+        response = self.client.get('/')
+        self.assertTemplateUsed(response, template_name='home.html')
+        self.assertIn('Welcome back {}! Continue being awesome!'.format(self.username),
+                      response.content)
+
+    def test_logged_in_home_links(self):
+        response = self.client.get('/')
+        self.assertIn('<a href="/">', response.content)
+        self.assertIn('<a href="/profile/">', response.content)
+        self.assertIn('<a href="/stream/">', response.content)
+        self.assertIn('<a href="/library/">', response.content)
+        self.assertIn('<a href="/accounts/logout/?next=/">', response.content)
```

Comment:
careful about being over-specific about the html your templates create.  Someday you add a class or ids to these links and your tests break.  Maybe just look for the correct URLs ?